command_line/anvil_correction.py

```diff
@@ -0,0 +1,325 @@
+# coding: utf-8
+
+"""
+Correct integrated intensities to account for attenuation by a diamond anvil cell.
+
+High pressure X-ray diffraction experiments often involve a diamond anvil pressure
+cell, in which the sample is sandwiched between two anvils, effectively parallel flat
+plates of diamond.  The passage of the incident and diffracted beam through the
+anvils results in attenuation of both beams by the diamond by an amount that is
+dependent on the path length of each beam through each anvil.
+
+This utility calculates these path lengths and boosts the integrated reflection
+intensities to remove the calculated effect of the diamond attenuation.
+
+It is intended that this program be used to correct reflection intensities after
+integration but before scaling.  Call it on the output of dials.integrate.
+
+Examples::
+
+  dials.anvil_correction integrated.expt integrated.refl
+
+  dials.anvil_correction integrated.expt integrated.refl thickness=1.2 normal=1,0,0
+"""
+
+from __future__ import absolute_import, division, print_function
+
+import logging
+import os
+import sys
+import warnings
+
+import numpy as np
+from scipy.spatial.transform import Rotation
+
+from cctbx.eltbx import attenuation_coefficient
+from dials.array_family import flex
+import dials.util.log
+from dials.util.multi_dataset_handling import (
+    parse_multiple_datasets,
+    sort_tables_to_experiments_order,
+)
+from dials.util.options import flatten_experiments, flatten_reflections, OptionParser
+from dxtbx.model import Experiment
+import libtbx.phil
+
+try:
+    from typing import List, Sequence, SupportsFloat
+except ImportError:
+    pass
+else:
+    Vector = Sequence[SupportsFloat]
+
+
+filename_root = os.path.splitext(os.path.basename(__file__))[0]
+logger = logging.getLogger("dials.%s" % filename_root)
```

Comment:
```suggestion
logger = logging.getLogger("dials.command_line.anvil_correction")
```
The `command_line` bit is missing. Doesn't make any difference, but is convention.